src/main.rs

```diff
@@ -0,0 +1,449 @@
+                .try_into()
+                .expect("Assumption is that at least two members will be allowed."),
+            ..Default::default()
+        },
+        price_aggregator: PriceAggregatorConfig {
+            circular_buffer: vec![CurrencyAmount::from_rational(5, 10)]
+                .try_into()
+                .expect("Must work since buffer should have at least a single value."),
+        },
+        council_membership: CouncilMembershipConfig {
+            members: account_ids
+                .clone()
+                .try_into()
+                .expect("Should support at least 5 members."),
+            phantom: Default::default(),
+        },
+        technical_committee_membership: TechnicalCommitteeMembershipConfig {
+            members: account_ids[..3]
+                .to_vec()
+                .try_into()
+                .expect("Should support at least 3 members."),
+            phantom: Default::default(),
+        },
+        community_council_membership: CommunityCouncilMembershipConfig {
+            members: account_ids
+                .try_into()
+                .expect("Should support at least 5 members."),
+            phantom: Default::default(),
+        },
+        council: Default::default(),
+        technical_committee: Default::default(),
+        community_council: Default::default(),
+        democracy: Default::default(),
+        treasury: Default::default(),
+        community_treasury: Default::default(),
+    };
+
+    config
+        .build_storage()
+        .expect("Genesis config should build successfully")
+}
+
+fn process_input(accounts: &[&GenesisAccount<Public>], genesis: &Storage, data: &[u8]) {
+    let mut extrinsic_data = data;
+
+    let extrinsics: Vec<(u8, u8, RuntimeCall)> =
+        iter::from_fn(|| DecodeLimit::decode_with_depth_limit(64, &mut extrinsic_data).ok())
+            .filter(|(_, _, x): &(_, _, RuntimeCall)| {
+                !recursively_find_call(x.clone(), call_filter)
+            })
+            .collect();
+
+    let mut block: u32 = 1;
+    let mut weight: Weight = Weight::zero();
+    let mut elapsed: Duration = Duration::ZERO;
+
+    BasicExternalities::execute_with_storage(&mut genesis.clone(), || {
+        let initial_total_issuance = TotalIssuance::<Runtime>::get();
+
+        initialize_block(block, None);
+
+        for (lapse, origin, extrinsic) in extrinsics {
+            if lapse > 0 {
+                let prev_header = finalize_block(elapsed);
+
+                // We update our state variables
+                block += u32::from(lapse);
+                weight = Weight::zero();
+                elapsed = Duration::ZERO;
+
+                // We start the next block
+                initialize_block(block, Some(&prev_header));
+            }
+
+            weight.saturating_accrue(extrinsic.get_dispatch_info().call_weight);
+            if weight.ref_time() >= 2 * WEIGHT_REF_TIME_PER_SECOND {
```

Comment:
You can fetch this value from the runtime config to avoid duplicating it.

Reply:
Both `MAXIMUM_BLOCK_WEIGHT` and `WEIGHT_REF_TIME_PER_SECOND` are private

Reply:
`RuntimeBlockWeights` should be public though - you can fetch all limit related values from their AFAIK.